src/Functions/array/arrayFirstLast.cpp

```diff
@@ -67,16 +87,29 @@ struct ArrayFirstLastImpl
                     else
                     {
                         out->insertDefault();
+
+                        if constexpr (element_not_exists_strategy == ArrayFirstLastElementNotExistsStrategy::Null)
+                            (*vec_null_map_to)[offset_index] = true;
                     }
                 }
 
+                if constexpr (element_not_exists_strategy == ArrayFirstLastElementNotExistsStrategy::Null)
+                    return ColumnNullable::create(std::move(out), std::move(col_null_map_to));
+
                 return out;
             }
             else
             {
                 auto out = array.getData().cloneEmpty();
-                out->insertDefault();
-                return out->replicate(IColumn::Offsets(1, array.size()));
+                out->insertManyDefaults(array.size());
+
+                if constexpr (element_not_exists_strategy == ArrayFirstLastElementNotExistsStrategy::Null)
+                {
+                    auto col_null_map_to = ColumnUInt8::create(out->size(), true);
+                    return ColumnNullable::create(std::move(out), std::move(col_null_map_to));
+                }
```

Comment:
Why we insert default before converting to nullable? Should we insert default or null here?

Reply:
Memory sanitizer will raise an error during access of uninitialized data when column serialization/deserialization will take place. It can be fixed to avoid performance penalty, but for this function it is okay.